core/src/test/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlanTestSuite.scala

```diff
@@ -120,14 +125,136 @@ class LogicalPlanTestSuite extends BasePlanTest {
     }
   }
 
+  // https://github.com/pingcap/tispark/issues/2290
+  test("fix cannot encode row key with non-long type") {
+    tidbStmt.execute("DROP TABLE IF EXISTS `t1`")
+    if (StoreVersion.minTiKVVersion("5.0.0", this.ti.tiSession.getPDClient)) {
+      tidbStmt.execute("""
+          |CREATE TABLE `t1` (
+          |  `a` BIGINT UNSIGNED  NOT NULL,
+          |  `b` varchar(255) NOT NULL,
+          |  `c` varchar(255) DEFAULT NULL,
+          |  PRIMARY KEY (`a`) CLUSTERED
+          |  ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin""".stripMargin)
+    } else {
+      if (isEnableAlterPrimaryKey) {
+        cancel("TiDB config alter-primary-key must be false")
+      }
+      tidbStmt.execute("""
+                         |CREATE TABLE `t1` (
+                         |  `a` BIGINT UNSIGNED  NOT NULL,
+                         |  `b` varchar(255) NOT NULL,
+                         |  `c` varchar(255) DEFAULT NULL,
+                         |  PRIMARY KEY (`a`)
+                         |  ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin""".stripMargin)
+    }
+    tidbStmt.execute(
+      " INSERT INTO t1 VALUES(0, 'aa', 'aa'), ( 9223372036854775807, 'bb', 'bb'), ( 9223372036854775808, 'cc', 'cc'), ( 18446744073709551615, 'dd', 'dd')")
+
+    def checkResultAndExpression(
+        sql: String,
+        valueExpectation: List[String],
+        expressionExpectation: Expression): Unit = {
+      val df = spark.sql(sql)
+      val dag = extractDAGRequests(df).head
+      val rangeFilter = dag.getRangeFilter
+      val value = df.collect()
+      for (x <- value.indices) {
```

Comment:
use scalatest matcher
`value should contain theSameElementsAs valueExpectation`

Reply:
ok